awx/main/dispatch/__init__.py

```diff
@@ -19,7 +19,17 @@ def get_task_queuename():
     if os.getenv('AWX_COMPONENT') == 'web':
         from awx.main.models.ha import Instance
 
-        return Instance.objects.filter(node_type__in=['control', 'hybrid']).order_by('?').first().hostname
+        return (
+            Instance.objects.filter(
+                node_type__in=['control', 'hybrid'],
+                node_state__in=['ready'],
+                enabled__in=['true'],
```

Comment:
This is just kind of weird, you should be able to do `enabled=True` for this kwarg, Likewise, `node_state='ready'`, although I'm skeptical that node_state should apply to these types of nodes

Reply:
yah mostly... symatry and figure later on its easier to append to the list that's why i did __in

Reply:
i will change it

Reply:
updated, and also i change 'control' and 'hybrid' to Instance.Types.CONTROL and Instance.Types.HYBRID